byzcoin/proof_test.go

```diff
@@ -23,36 +23,39 @@ func TestNewProof(t *testing.T) {
 	s := createSC(t)
 	p, err := NewProof(s.c, s.s, skipchain.SkipBlockID{}, []byte{})
 	require.NotNil(t, err)
-	p, err = NewProof(s.c, s.s, s.genesis.Hash, []byte{1})
+
+	key := []byte{1}
+	p, err = NewProof(s.c, s.s, s.genesis.Hash, key)
 	require.Nil(t, err)
-	require.False(t, p.InclusionProof.Match())
+	require.False(t, p.InclusionProof.Match(key))
```

Comment:
see the PR description

Reply:
OK - I'm split on this one. I see your point, but it makes using the inclusionproof more difficult. But there is some gain in security by default, because you _have_ to check the key.

So +1 for keeping the key-argument.